ferry.go

```diff
@@ -596,7 +595,7 @@ func (f *Ferry) Run() {
 			if ctx.Err() == nil {
 				// Ghostferry is still running
 				if f.OverallState.Load() != StateCutover {
-					// Save state dump and exit if not during the cutover stage
+					f.DataIterator.WaitForGracefulShutdown()
```

Comment:
The graceful shutdown of iterators was introduced to guarantee that the shutdown state dump contain all copied rows. Most tests also rely on shutting down the process after one data iteration, which will trigger when one goroutine is finished, making it difficult to assert states when another goroutine is half-way.

Reply:
Are these golang tests? I think in the ruby tests we may be able to block the data iterator.

Also: we might want to introduce ways to force per-table-parallelism to be 1, not only for tests, but perhaps also for actual runs. This might help with some sort of phased rollout of this code.

Reply:
When thinking about this again, I think we should get rid of the graceful shutdown and let rows be copied again on resume. And as you say, fix tests and code to allow parallelism to be turned off easily.